__device-tests__/utils.js

```diff
@@ -157,3 +157,98 @@ export async function takeScreenshot( { withoutKeyboard, crop } = {} ) {
 }
+
+/**
+ * Generates the appropriate URL to fetch the theme data based on the provided theme details.
+ *
+ * @param {Object}  theme                  Object containing details about the theme.
+ * @param {string}  theme.name             The name of the theme.
+ * @param {boolean} theme.isWordPressTheme Flag indicating whether the theme is a WordPress theme. Defaults to false.
+ * @return {string} The URL from which the theme data can be fetched.
+ */
+function getThemeLink( { name, isWordPressTheme = false } ) {
+	if ( isWordPressTheme ) {
+		return `https://raw.githubusercontent.com/WordPress/${ name }/trunk/theme.json`;
+	}
+
+	return `https://raw.githubusercontent.com/Automattic/themes/trunk/${ name }/theme.json`;
+}
+
+/**
+ * Fetches and processes the theme data from the provided theme details.
+ *
+ * @param {Object}  theme                  Object containing details about the theme.
+ * @param {string}  theme.name             The name of the theme.
+ * @param {boolean} theme.isWordPressTheme Flag indicating whether the theme is a WordPress theme. Defaults to false.
+ * @return {Promise<Object>} A promise that resolves to a stringified JSON object containing the theme data.
+ * @throws Will throw an error if the fetch operation fails.
+ */
+export async function fetchTheme( theme ) {
+	const themeJSONLink = getThemeLink( theme );
+
+	return await fetch( themeJSONLink )
+		.then( ( response ) => response.json() )
+		.then( ( data ) => {
+			const rawFeatures = data?.settings;
+			const rawStyles = data?.styles;
+			const supportedBlocks = [
+				'core/paragraph',
+				'core/button',
+				'core/quote',
+				'core/pullquote',
+				'core/quote',
+				'core/search',
+			];
+			const supportedElements = [
+				'h1',
+				'h2',
+				'h3',
+				'h4',
+				'h5',
+				'h6',
+				'link',
+				'button',
+			];
+
+			if ( rawFeatures?.color?.palette ) {
+				rawFeatures.color.palette = {
+					theme: rawFeatures.color.palette,
+				};
+			}
+			if ( rawFeatures.typography.fontSizes ) {
+				rawFeatures.typography.fontSizes = {
+					theme: rawFeatures.typography.fontSizes,
+				};
+			}
+
+			// We filter the supported features because of limitations in sending this data
+			// during app launch on Android.
+			const filteredRawFeatures = {
+				color: rawFeatures?.color,
+				typography: {
+					fontSizes: rawFeatures?.typography?.fontSizes,
+				},
+			};
+			const filteredRawStyles = {
+				color: rawStyles?.color,
+				typography: rawStyles?.typography,
+				blocks: supportedBlocks.reduce( ( acc, key ) => {
+					if ( rawStyles?.blocks?.[ key ] ) {
+						acc[ key ] = rawStyles.blocks[ key ];
+					}
+					return acc;
+				}, {} ),
+				elements: supportedElements.reduce( ( acc, key ) => {
+					if ( rawStyles?.elements?.[ key ] ) {
+						acc[ key ] = rawStyles.elements[ key ];
+					}
+					return acc;
+				}, {} ),
+			};
+
+			return {
+				rawFeatures: JSON.stringify( filteredRawFeatures ),
+				rawStyles: JSON.stringify( filteredRawStyles ),
```

Comment:
These values will be stringified when being passed when launching the demo app ([reference](https://github.com/WordPress/gutenberg/blob/7652de9ed159d4e9698e432c06b83e56777160c8/packages/react-native-editor/__device-tests__/helpers/utils.js#L758-L780)). I'm wondering if we could simply return the original values and let the data transformation happen when initializing the editor. WDYT?

Reply:
The thing is that the editor expects these data to be string values, so if we remove this and just keep the `JSON.stringify` in `launchApp` the native side will get `rawFeatures` and `rawStyles` as JSON objects, then we will need to convert them into strings again. So I thought I'd simplify that whole process 😅

Reply:
Ah, ok I see. I didn't realize that the editor expects this data as strings 👍.